packages/react-codemirror/src/e2e_tests/autoCompletion.spec.tsx

```diff
@@ -244,13 +244,42 @@ test('can complete pattern snippet', async ({ page, mount }) => {
   await page.locator('.cm-tooltip-autocomplete').getByText('-[]->()').click();
   await expect(page.locator('.cm-tooltip-autocomplete')).not.toBeVisible();
 
-  await textField.press('Tab');
   await textField.press('Tab');
   await textField.press('Tab');
 
   await expect(textField).toHaveText('MATCH ()-[]->()-[ ]->( )');
 });
 
+test('can accept completion inside pattern snippet', async ({
+  page,
+  mount,
+}) => {
+  await mount(<CypherEditor schema={{ labels: ['City'] }} />);
+  const textField = page.getByRole('textbox');
+
+  await textField.fill('MATCH ()-[]->()');
+
+  await page.locator('.cm-tooltip-autocomplete').getByText('-[]->()').click();
+  await expect(page.locator('.cm-tooltip-autocomplete')).not.toBeVisible();
+
+  // move to node
+  await textField.press('Tab');
+
+  // get & accept completion
+  await textField.press(':');
+  await expect(
+    page.locator('.cm-tooltip-autocomplete').getByText('City'),
+  ).toBeVisible();
+
+  await textField.press('Tab');
+  await expect(page.locator('.cm-tooltip-autocomplete')).not.toBeVisible();
+
+  // tab out of the snippet
+  await textField.press('Tab');
+
+  await expect(textField).toHaveText('MATCH ()-[]->()-[ ]->(:City)');
+});
+
```

Comment:
Do we have any test to check the shift + Tab traverses the snippets back or that the Escape clears them out?

Reply:
No, I but I'll add some :+1:

Reply:
I broke out a `snippets.spec.tsx` file and added a test for navigating snippets